runner/utils/driver/docker/docker.go

```diff
@@ -0,0 +1,255 @@
+	if err != nil {
+		logrus.Errorf("error applying service in docker unable to load host file with suitable default - %v", err)
+		return err
+	}
+
+	service.Labels["internalRuntime"] = string(service.Runtime)
+
+	for _, task := range service.Tasks {
+		// todo get image
+		// for now images have been created locally but not uploaded on docker hub
+		out, err := d.client.ImagePull(ctx, task.Docker.Image, types.ImagePullOptions{})
+		if err != nil {
+			return err
+		}
+		io.Copy(os.Stdout, out)
+
+		// TODO check docker runtime field in host config
+		// expose ports of docker container as specified in task
+		exposedPorts := map[nat.Port]struct{}{}
+		for _, port := range task.Ports {
+			portString := strconv.Itoa(int(port.Port))
+			portWithProtocol := nat.Port(fmt.Sprintf("%s/%s", portString, port.Protocol))
+			exposedPorts[portWithProtocol] = struct{}{}
+		}
+
+		// set environment variables of docker container
+		task.Env["URL"] = fmt.Sprintf("http://service.artifact.svc.cluster.local:4122")
+		token, err := d.auth.GenerateHS256Token(service.ID, service.ProjectID, service.Version)
+		if err != nil {
+			logrus.Errorf("error applying service in docker unable generate token - %v", err)
+			return err
+		}
+		task.Env["TOKEN"] = token
+		envs := []string{}
+		for envName, envValue := range task.Env {
+			envs = append(envs, fmt.Sprintf("%s=%s", envName, envValue))
+		}
+		//
+		// service.Labels["internalServiceId"] = service.ID
+		// service.Labels["internalProjectId"] = service.ProjectID
+
+		containerName := fmt.Sprintf("%s--%s--%s--%s", service.ProjectID, service.ID, task.ID, service.Version)
+		resp, err := d.client.ContainerCreate(ctx, &container.Config{
+			Image:        task.Docker.Image,
+			Env:          envs,
+			ExposedPorts: exposedPorts,
+			Labels:       service.Labels,
+		}, &container.HostConfig{
+			// receiving memory in mega bytes converting into bytes
+			// convert received mill cpus to cpus by diving by 1000 then multiply with 100000 to get cpu quota
+			Resources: container.Resources{Memory: task.Resources.Memory * 1024 * 1024, CPUQuota: task.Resources.CPU * 100},
+		}, nil, containerName)
+		if err != nil {
+			logrus.Errorf("error applying service in docker unable to create container %s got error message - %v", containerName, err)
+			return err
+		}
+
+		if err := d.client.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{}); err != nil {
+			logrus.Errorf("error applying service in docker unable to start container %s got error message - %v", containerName, err)
+			return err
+		}
+
+		// get ip address of service & store it in host file
+		data, err := d.client.ContainerInspect(ctx, resp.ID)
+		if err != nil {
+			logrus.Errorf("error applying service in docker unable to inspect container %s got error message  -%v", containerName, err)
+			return err
+		}
+		hostFile.AddHost(data.NetworkSettings.IPAddress, fmt.Sprintf("%s.%s-%s.svc.cluster.local", service.ID, service.ProjectID, task.ID))
```

Comment:
the fromat is `serviceID.projectID.svc.cluster.local`. There is no task id in the mix